ray-curator/tests/stages/test_filters.py

```diff
@@ -0,0 +1,1320 @@
+            data=pd.DataFrame({"documents": ["Five aaa aa", "aaaSeven aaaa"]}),
+            task_id="batch_1_letter_count",
+            dataset_name="test_1",
+        )
+        expected_data.data[score_field] = pd.Series([5, 7])
+        assert all_equal(expected_data, filtered_data), f"Expected {expected_data} but got {filtered_data}"
+
+    def test_filter_document(self, letter_count_data: DocumentBatch) -> None:
+        letter_filter = LetterCountFilter()
+        score_field = "a_count"
+        score_step = Score(
+            letter_filter.score_document,
+            text_field="documents",
+            score_field=score_field,
+        )
+
+        score_step.setup()
+        scored_data = score_step.process_batch([letter_count_data])[0]
+
+        filter_step = Filter(letter_filter.keep_document, score_field)
+
+        filter_step.setup()
+        filtered_data = filter_step.process_batch([scored_data])[0]
+
+        expected_data = DocumentBatch(
+            data=pd.DataFrame({"documents": ["Five aaa aa", "aaaSeven aaaa"]}),
+            task_id="batch_1_score_fn_filter_fn",
+            dataset_name="test_1",
+        )
+        expected_data.data[score_field] = pd.Series([5, 7])
+        assert all_equal(expected_data, filtered_data), f"Expected {expected_data} but got {filtered_data}"
+
+    def test_filter(self, letter_count_data: DocumentBatch) -> None:
+        letter_filter = LetterCountFilter()
+        score_field = "a_count"
+        score_step = Score(
+            letter_filter,
+            text_field="documents",
+            score_field=score_field,
+        )
+
+        score_step.setup()
+        scored_data = score_step.process_batch([letter_count_data])[0]
+
+        filter_step = Filter(letter_filter, score_field)
+
+        filter_step.setup()
+        filtered_data = filter_step.process_batch([scored_data])[0]
+
+        expected_data = DocumentBatch(
+            data=pd.DataFrame({"documents": ["Five aaa aa", "aaaSeven aaaa"]}),
+            task_id="batch_1_score_fn_filter_fn",
+            dataset_name="test_1",
+        )
+        expected_data.data[score_field] = pd.Series([5, 7])
+        assert all_equal(expected_data, filtered_data), f"Expected {expected_data} but got {filtered_data}"
+
+    def test_invert(self, letter_count_data: DocumentBatch) -> None:
+        letter_filter = LetterCountFilter()
+        filter_step = ScoreFilter(letter_filter, text_field="documents", invert=True)
+
+        filter_step.setup()
+        filtered_data = filter_step.process_batch([letter_count_data])[0]
+
+        expected_data = DocumentBatch(
+            data=pd.DataFrame({"documents": ["Two aa", "a a Three a"]}),
+            task_id="batch_1_letter_count",
+            dataset_name="test_1",
+        )
+        assert all_equal(expected_data, filtered_data), f"Expected {expected_data} but got {filtered_data}"
+
+    def test_score_batch_size(self, letter_count_data: DocumentBatch) -> None:
```

Comment:
I added the `test_score_batch_size`, `test_filter_batch_size`, and `test_score_filter_batch_size` specifically as tests for processing more than 1 `DocumentBatch` at a time. Other than that, all tests in this file already existed.